### PR TITLE
rename client's proxy option

### PR DESCRIPTION
**Pull request by AlexanderZagaynov**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1623862

In a recent `google-api-ruby-client` gem versions option `proxy` now named as `proxy_url`:
https://github.com/googleapis/google-api-ruby-client/blob/8faa43b9f7e6b06cc820df4bcffb384b988e33d1/lib/google/apis/options.rb#L21

This PR align that option name accordingly.

P.S. There is one more PR for that: https://github.com/fog/fog-google/pull/418
but current one can be merged without waiting.

### CODE REVIEW

**Comment by AlexanderZagaynov** [comment]
@tumido @juliancheal please review

**Comment by juliancheal** [comment]
This PR looks great @AlexanderZagaynov. Although I'm not sure what relation it has the the fog-google PR. Sorry for misunderstanding.

**Comment by AlexanderZagaynov** [comment]
@juliancheal your previous commit to `fog-google` was borked by some update (I've put information in that PR).
Those two PRs together will solve mentioned BZ issue.

**Comment by juliancheal** [comment]
@AlexanderZagaynov ah that makes sense.

@tumido I thought if we merge this one, at least our code is correct?

**Comment by AlexanderZagaynov** [comment]
@juliancheal Tom's proposal to wait for `fog-google` gem update makes sense.

**Comment by JPrause** [comment]
@miq-bot add_label blocker

**Comment by JPrause** [comment]
@AlexanderZagaynov if this can be backported, can you add the gaprindashvili/yes label.

**Comment by AlexanderZagaynov** [comment]
@miq-bot add_label gaprindashvili/yes

**Comment by AlexanderZagaynov** [comment]
@tumido @juliancheal `fog-google` PR was merged, however wasn't released as a version yet.
Gem maintainer (Artem @Temikus) told me that he going to release it in a days, so I just fixed that gem version in our `gemspec`. We can wait merge till `fog-google` release, or we can simply point it to github repo, if we're in hurry.

File: manageiq-providers-google.gemspec [hunk 1]
```diff
@@ -13,7 +13,7 @@ Gem::Specification.new do |s|
 
   s.files = Dir["{app,config,lib}/**/*"]
 
-  s.add_dependency "fog-google", ">=1.5.0"
+  s.add_dependency "fog-google", ">= 1.8.0", "< 2.0.0"
```
Comment (agrare) [inline manageiq-providers-google.gemspec]: This can just be `~> 1.8`

**Comment by agrare** [comment]
I think we're going to have a problem with moving to the newer fog-google, on master they have `spec.add_dependency "fog-core", ">= 2.0", "<= 2.1.0"` where on v1.7.1 they just have `spec.add_dependency "fog-core"`.

We're currently using fog-core 1.45 I believe and are in the process of updating other provider gems to support fog-core 2.* but were going to wait until after hammer was branched to merge these.

This would require we pull those back to gaprindashvilli.

Ideally the fog-google maintainer could pull this change to v1.7.2 and leave the gemspec requirements the same.

File: manageiq-providers-google.gemspec (hunk 1 above)
Comment (AlexanderZagaynov) [reply manageiq-providers-google.gemspec]: yep

**Comment by AlexanderZagaynov** [comment]
@agrare I discussed that with Artem, and he told me, he is going to do that in from v2.0 only

**Comment by agrare** [comment]
Okay awesome thanks @AlexanderZagaynov

**Comment by AlexanderZagaynov** [comment]
@tumido @juliancheal @agrare done, can merge now

**Comment by agrare** [comment]
Awesome thanks a lot @Temikus!  We're upgrading to newer fog-core in this release but need this fix on older releases.  Really appreciate your help.

**Comment by juliancheal** [comment]
Well done on getting it all committed in fog too!

**Comment by AlexanderZagaynov** [comment]
@miq-bot add_label hammer/yes

**Comment by simaishi** [comment]
Hammer backport details:

```
$ git log -1
commit af5dfef52fe510f849844083b0cbb2f9c244738f
Author: Adam Grare <agrare@redhat.com>
Date:   Thu Oct 4 09:16:31 2018 -0400

    Merge pull request #73 from AlexanderZagaynov/BZ-1623862_http_proxy
    
    rename client's proxy option
    
    (cherry picked from commit ec11cad1b0d230581173c19f9111fb9613eb1dbe)
    
    Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1623862
```

**Comment by simaishi** [comment]
As per BZ Target Release, removing `gaprindashvili/yes`

**Pull request merged**